### PR TITLE
feat(diagram-converter): flag generated forms for manual migration

### PR DESCRIPTION
**Pull request by bojtospeter**
## Motivation

Camunda 7 Generated Task Forms cannot be transformed into Camunda 8 forms automatically. The Diagram Converter should preserve its existing cleanup behavior while giving model owners one actionable manual-migration task.

This is stack layer **1/2** and targets `main`. The dependent migration-skill PR completes the issue, so this PR intentionally does not close it.

## Changes

- Emit exactly one owner-level `TASK` finding with message ID `form-data` for a user task or start event containing `camunda:formData` or legacy direct `camunda:formProperty` elements.
- Keep mixed `formData`/`formProperty` and duplicate malformed `formData` blocks deduplicated to one finding per owner.
- Do not generate Camunda 8 forms; continue removing Camunda 7 generated-form metadata from converted BPMN.
- Point the manual migration task to the official Camunda 8 forms documentation.
- Register the owner visitors and update the message template.

## Tests

- Add a category-scoped parameterized Java test covering form data, direct form properties, user tasks, start events, mixed metadata, and duplicate form-data blocks.
- Add YAML conversion coverage for legacy direct form properties on user tasks and start events, including converted BPMN and task findings.

## Compatibility

Existing generated-form metadata removal remains unchanged. The only output addition is the deduplicated manual-migration finding.

## Baseline and validation

Baseline commit: `7f8891abc743fa0caeae788da672382ca733e111`

Passed with Java 21:

```text
cd diagram-converter && JAVA_HOME=/opt/homebrew/opt/openjdk@21 PATH=/opt/homebrew/opt/openjdk@21/bin:$PATH mvn verify -Pdistro,checkFormat --batch-mode --no-transfer-progress
```

related to #2326

### CODE REVIEW

File: diagram-converter/core/src/test/java/io/camunda/migration/diagram/converter/visitor/impl/element/GeneratedFormDataVisitorTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            "FormPropertyStart",
+            "formProperty",
+            """
+            <bpmn:startEvent id="FormPropertyStart">
+              <bpmn:extensionElements>
+                <camunda:formProperty id="name" type="string"/>
+                <camunda:formProperty id="approved" type="boolean"/>
+              </bpmn:extensionElements>
+            </bpmn:startEvent>
+            """),
```
[the remark this answers is not in the document]
Comment (bojtospeter) [reply diagram-converter/core/src/test/java/io/camunda/migration/diagram/converter/visitor/impl/element/GeneratedFormDataVisitorTest.java:126]: Good catch — fixed in `65a4b27c`. The category-scoped parameterized test now includes a start event with `camunda:formData`, covering both generated-form metadata shapes for start events.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: diagram-converter/core/src/main/resources/message-templates.properties [hunk 2]
```diff
@@ -179,8 +179,9 @@ failed-job-retry-time-cycle-error.severity=WARNING
 error-event-definition.message={{ templates.element-not-transformable-prefix }}
 error-event-definition.severity=WARNING
 #
-form-data.message={{ templates.element-not-transformable-prefix }} Please define a form key instead.
+form-data.message={{ templates.element-not-transformable-prefix }} Migrate this Generated Task Form to a Camunda 8 form and link it manually.
 form-data.severity=TASK
+form-data.link=https://docs.camunda.io/docs/components/modeler/forms/utilizing-forms/
```
[the remark this answers is not in the document]
Comment (bojtospeter) [reply diagram-converter/core/src/main/resources/message-templates.properties:184]: The wording is intentional and matches the issue requirement exactly: the finding must state that the “Generated Task Form” cannot be transformed, including when the legacy form metadata belongs to a start event. Changing it to a neutral term would diverge from the specified user-facing text, so I’m retaining the current template.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).
